controllers/issuer_controller.go

```diff
@@ -75,6 +82,7 @@ func (r *IssuerReconciler) Reconcile(ctx context.Context, req ctrl.Request) (res
 	logger.V(2).Info("Starting reconcile loop", "name", req.Name, "namespace", req.Namespace)
 
 	result, issuerStatusPatch, returnedError := r.reconcileStatusPatch(logger, ctx, req)
```

Comment:
**Question:** Why is `returnedError` left unchecked? That feels unexpected. If that's expected, then it should be explained in a comment IMO, e.g.

```suggestion
	// The error returned by `reconcileStatusPatch` is meant for controller-runtime's
	// reconciliation logic and is meant to be returned unchecked. That's why we aren't
	// checking `returnedError != nil` .
	result, issuerStatusPatch, returnedError := r.reconcileStatusPatch(logger, ctx, req)
```